webapp/src/components/sidebar/boardTemplateMenuItem.tsx

```diff
@@ -39,8 +32,49 @@ const addBoardFromTemplate = async (intl: IntlShape, showBoard: (id: string) =>
     }
 }
 
+type ButtonProps = {
+    showBoard: (id: string) => void
+    boardTemplate: Board
+}
+
+export const BoardTemplateMenuItemOptionsButton = React.memo((props: ButtonProps) => {
+    const intl = useIntl()
+    const {showBoard, boardTemplate} = props
+
+    return (
+        <MenuWrapper stopPropagationOnToggle={true}>
+            <IconButton icon={<OptionsIcon/>}/>
+            <Menu position='right'>
+                <Menu.Text
+                    icon={<EditIcon/>}
+                    id='edit'
+                    name={intl.formatMessage({id: 'Sidebar.edit-template', defaultMessage: 'Edit'})}
+                    onClick={() => {
+                        showBoard(boardTemplate.id || '')
+                    }}
+                />
+                <Menu.Text
+                    icon={<DeleteIcon/>}
+                    id='delete'
+                    name={intl.formatMessage({id: 'Sidebar.delete-template', defaultMessage: 'Delete'})}
+                    onClick={async () => {
+                        await mutator.deleteBlock(boardTemplate, 'delete board template')
```

Comment:
Should there be a confirmation dialogue for this? I know there's another issue where someone implemented it. @asaadmahmood can confirm if we're adopting this design.

Reply:
If we want to do this, lets create a new ticket for it since it will further delay this ticket. 
I'll skip the implementation for the sake of getting this ticket out to the public sooner.